evap/results/views.py

```diff
@@ -63,13 +63,17 @@ def course_detail(request, semester_id, course_id):
     sections = calculate_results(course)
 
     if request.user.is_reviewer:
-        public_view = request.GET.get('public_view') != 'false'  # if parameter is not given, show public view.
+        view = request.GET.get('view', 'public')  # if parameter is not given, show public view.
     else:
-        public_view = request.GET.get('public_view') == 'true'  # if parameter is not given, show own view.
+        view = request.GET.get('view', 'full')  # if parameter is not given, show own view.
+
+    contributor_id = request.user.id
+    if view == 'filter' and user.is_staff:
```

Comment:
`user` is not defined here (should be `request.user`)

Reply:
Maybe we should add a test? Something like this should trigger a test failure imo.